src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,638 @@
+    );
+
+    // function called by Thumb keyboard processing
+    const setThumbValue = React.useCallback(
+      (index: number, value: number) => {
+        const newValues = [...currentValues];
+        newValues[index] = value;
+        setCurrentValues(newValues);
+        onChange && onChange(newValues);
+      },
+      [currentValues, onChange],
+    );
+
+    const setThumbMoveIndex = React.useCallback((index: number) => {
+      setActiveMoveIndex(index);
+    }, []);
+
+    const handlePointerUp = React.useCallback(
+      (event: PointerEvent) => {
+        updateThumbValue(event, onChange);
+        setActiveMoveIndex(undefined);
+        event.preventDefault();
+        event.stopPropagation();
+      },
+      [onChange, updateThumbValue],
+    );
+
+    const handlePointerDownOnSlider = React.useCallback(
+      (event: React.PointerEvent) => {
+        if (containerRef.current) {
+          const percent = getPercentageOfRectangle(
+            containerRef.current.getBoundingClientRect(),
+            event.pageX,
+          );
+          let pointerValue =
+            railDomain.min + (railDomain.max - railDomain.min) * percent;
+          pointerValue = roundValueToClosestStep(pointerValue, step);
+
+          const closestValueIndex = getClosestValueIndex(
+            currentValues,
+            pointerValue,
+          );
+          const [minVal, maxVal] = getAllowableThumbRange(
+            thumbMode,
+            currentValues,
+            closestValueIndex,
+            railDomain,
+            step,
+          );
+
+          pointerValue = clamp(pointerValue, minVal, maxVal);
+          const newValues = [...currentValues];
+          newValues[closestValueIndex] = pointerValue;
+          setCurrentValues(newValues);
+          onChange && onChange(newValues);
+        }
+      },
+      [railDomain, step, currentValues, thumbMode, onChange],
+    );
+
+    React.useEffect(() => {
+      const ownerDoc = containerRef.current?.ownerDocument;
+      ownerDoc?.addEventListener('pointermove', handlePointerMove);
+      ownerDoc?.addEventListener('pointerup', handlePointerUp);
+      return () => {
+        ownerDoc?.removeEventListener('pointermove', handlePointerMove);
+        ownerDoc?.removeEventListener('pointerup', handlePointerUp);
+      };
+    }, [handlePointerMove, handlePointerUp]);
+
+    const componentContainerClassNames = cx(
+      'iui-slider-component-container',
+      disabled && 'iui-disabled',
+      containerClassName,
+    );
+
+    return (
+      <SliderContext.Provider
+        value={{
+          values: currentValues,
+          trackDisplayMode: trackDisplay,
+          toolTipFunction,
+          railDomain,
+          setThumbMoveIndex,
+          setThumbValue,
+          step,
+          thumbMode,
+          toolTipPosition,
+          hideTooltip,
+        }}
+      >
+        <div
+          className={componentContainerClassNames}
+          {...remainingContainerProps}
```

Comment:
No need for two separate consts. Just use containerProps directly
```suggestion
		  {..containerProps}
          className={cx(
            'iui-slider-component-container',
	        disabled: 'iui-disabled',
	        containerProps.className,
	      )}
```